docs/development.md

```diff
@@ -0,0 +1,140 @@
+
+Examples can be found in the [advection example](../example/advection/advection_package.cpp)
+```diff
+-  Real vmin = 1.0;
+-  Real vmax = 0.0;
+-  for (int k = kb.s; k <= kb.e; k++) {
+-    for (int j = jb.s; j <= jb.e; j++) {
+-      for (int i = ib.s; i <= ib.e; i++) {
+-        vmin = (v(k, j, i) < vmin ? v(k, j, i) : vmin);
+-        vmax = (v(k, j, i) > vmax ? v(k, j, i) : vmax);
+-      }
+-    }
+-  }
++
++  typename Kokkos::MinMax<Real>::value_type minmax;
++  Kokkos::parallel_reduce(
++      "advection check refinement",
++      Kokkos::MDRangePolicy<Kokkos::Rank<3>>(pmb->exec_space, {kb.s, jb.s, ib.s},
++                                             {kb.e + 1, jb.e + 1, ib.e + 1},
++                                             {1, 1, ib.e + 1 - ib.s}),
++      KOKKOS_LAMBDA(int k, int j, int i,
++                    typename Kokkos::MinMax<Real>::value_type &lminmax) {
++        lminmax.min_val = (v(k, j, i) < lminmax.min_val ? v(k, j, i) : lminmax.min_val);
++        lminmax.max_val = (v(k, j, i) > lminmax.max_val ? v(k, j, i) : lminmax.max_val);
++      },
++      Kokkos::MinMax<Real>(minmax));
+```
+(note the explicit use of `pmb->exec_space` to use to execution space associated with the `MeshBlock`)
+or in the [buffer packing]() functions
+```diff
+-void PackData(ParArrayND<T> &src, T *buf, int sn, int en, int si, int ei, int sj, int ej,
+-              int sk, int ek, int &offset) {
+-  for (int n = sn; n <= en; ++n) {
+-    for (int k = sk; k <= ek; k++) {
+-      for (int j = sj; j <= ej; j++) {
+-#pragma omp simd
+-        for (int i = si; i <= ei; i++)
+-          buf[offset++] = src(n, k, j, i);
+-      }
+-    }
+-  }
++void PackData(ParArray4D<T> &src, ParArray1D<T> &buf, int sn, int en, int si, int ei,
++              int sj, int ej, int sk, int ek, int &offset, MeshBlock *pmb) {
++  int ni = ei + 1 - si;
++  int nj = ej + 1 - sj;
++  int nk = ek + 1 - sk;
++  int nn = en + 1 - sn;
++
++  pmb->par_for(
++      "PackData 4D", sn, en, sk, ek, sj, ej, si, ei,
++      KOKKOS_LAMBDA(const int n, const int k, const int j, const int i) {
++        buf(offset + i - si + ni * (j - sj + nj * (k - sk + nk * (n - sn)))) =
++            src(n, k, j, i);
++      });
++  offset += nn * nk * nj * ni;
++  return;
+```
+Note the explicit calculation of the offset within the kernel and the explicit increment of the offset by the full extent after the kernel.
+
+
+
+### FAQ
+
+- What's the difference between `GetDim` and `extent`?
+
+`ParArrayND` offer `GetDim` to access the underlying array dimension.
+Here, `GetDim(0)` refers to the "first" dimension (e.g., x-direction).
+`ParArray#D`s (with `#` being 1, 2, 3, ...) are direct typedefs to `Kokkos::View`s.
+Thus, a call to `extent(0)` returns the dimension along the first index.
+Given that `ParArray#D`s are constructed using reverse indices (note the `k,j,i` order in accessing elements), `extent` and `GetDim` using the same number usually have different meaning.
+
```

Comment:
I expected that this was where it came from and don't think that it's a problem (but we could discuss it during the next call).
Especially as the terminology is unique (dim vs extent) there's a clear separation. I just pointed it out because I stumbled across it (despite better knowledge).